### ISSUE TITLE
Upgrade to new invoker API

### ISSUE THREAD

**Issue by richorama** [OrleansContrib/OrleansDashboard#63]
...from build:

```
GrainProfiler.cs(30,37): warning CS0618: 'IProviderRuntime.GetInvokeInterceptor()' is obsolete: 'Retrieve InvokeInterceptor instances from the ServiceProvider property.' [C:\projects\orleansdashboard\OrleansDashboard\OrleansDashboard.csproj]
GrainProfiler.cs(31,13): warning CS0618: 'IProviderRuntime.SetInvokeInterceptor(InvokeInterceptor)' is obsolete: 'Register InvokeInterceptor instances with the service provider during configuration.' 
```

**Comment by rikbosch** [comment]
This is not possible when using a bootstrapperprovider until the silobuilder issue is completed (https://github.com/dotnet/orleans/issues/2936).

see https://gitter.im/dotnet/orleans?at=59635ad197cedeb0482623f3

**Comment by richorama** [comment]
@rikbosch it could be placed in the extension method:

https://github.com/OrleansContrib/OrleansDashboard/blob/master/OrleansDashboard/ExtensionMethods.cs#L17

**Comment by rikbosch** [comment]
You don't have access to the servicecollection, which is where the `IGrainCallFilter` should be added. From what I know, the only way is to register a custom StartupType. The dashboard should not override the custom startuptype.

You could add an extensionmethod to use inside the `ConfigureServices` method, but then the dashboard can' t be configured using XML configuration.

**Comment by SebastianStehle** [comment]
Solved in the 2.0 Branch
